ppa/archive/management/commands/hathi_images.py

```diff
@@ -0,0 +1,165 @@
+        if not output_dir.is_dir():
+            raise CommandError(
+                f"Output directory '{output_dir}' does not exist or is not a directory"
+            )
+
+        # use ids specified via command line when present
+        htids = kwargs.get("htids", [])
+
+        # by default, sync data for all non-suppressed hathi source ids
+        digworks = DigitizedWork.objects.filter(
+            status=DigitizedWork.PUBLIC, source=DigitizedWork.HATHI
+        )
+
+        # if htids are specified via parameter, use them to filter
+        # the queryset, to ensure we only sync records that are
+        # in the database and not suppressed
+        if htids:
+            digworks = digworks.filter(source_id__in=htids)
+
+        # bail out if there's nothing to do
+        # (e.g., explicit htids only and none valid)
+        if not digworks.exists():
+            self.stdout.write("No records to download; stopping")
+            return
+
+        # setup main progress bar
+        overall_progress = None
+        if self.options["progress"]:
+            overall_progress = progressbar.ProgressBar(
+                redirect_stdout=True, max_value=digworks.count(), max_error=False
+            )
+            overall_progress.start()
+
+        self.stdout.write(
+            f"Downloading images for {digworks.count()} record{pluralize(digworks)}"
+        )
+
+        for digwork in digworks:
+            vol_id = digwork.source_id
+            
+            # Determine output volume & thumbnail directories (create as needed)
+            vol_dir = output_dir / get_vol_dir(vol_id)
+            vol_dir.mkdir(parents=True, exist_ok=True)
+            thumbnail_dir = vol_dir / "thumbnails"
+            thumbnail_dir.mkdir(exist_ok=True)
+            
+            # Get filename-friendly version of htid
+            clean_htid = encode_htid(vol_id)
+            
+            # Determine page range
+            if digwork.item_type == DigitizedWork.FULL:
+                page_range = range(1, digwork.page_count+1)
+            else:
+                page_range = digwork.page_span
+            
+            # Setup volume-level progress bar
+            volume_progress = None
+            if self.options["progress"]:
+                volume_progress = progressbar.ProgressBar(
+                    redirect_stdout=True, max_value=len(page_range), max_error=False
+                )
+                volume_progress.start()
+
+            # Fetch images
+            stats = {
+                "image": {"fetch": 0, "skip": 0},
+                "thumbnail": {"fetch": 0, "skip": 0}
+            }
+            for page_num in page_range:
+                image_name = f"{clean_htid}.{page_num:08d}.jpg"
+
+                # Fetch thumbnail if file does not exist
+                page_thumbnail = thumbnail_dir / image_name
+                if not page_thumbnail.is_file():
+                    thumbnail_url = page_image_url(vol_id, page_num, 250)
```

Comment:
I suggest setting the image width as class variables so they are easier to find at the top of the file

Reply:
Maybe I'll shift more of that logic into `page_image_url`